src/foreign/core/cell.rs

```diff
@@ -13,7 +13,12 @@ where
 
     #[inline]
     fn size_hint(depth: usize) -> (usize, Option<usize>) {
-        <A as Arbitrary<'a>>::size_hint(depth)
+        Self::faillible_size_hint(depth).unwrap_or_default()
+    }
```

Comment:
The old `size_hint` can be default-implemented with `fallible_size_hint` or the default right? No need to repeat this incantation in every implementation.

Reply:
Not really,  because of backwards compatibility.

If we make the default `size_hint` delegate to `faillible_size_hint`, this means we can't provide a default implementation for `faillible_size_hint` that calls back to `size_hint`, otherwise there would be an infinite loop.

We *could* have a default `faillible_size_hint` implementation that returns `Ok((0, None))`,  but then, when libraries migrate to calling `faillible_size_hint`, it will break for out of date dependencies that only implement `size_hint`.